lib/ocpp/v201/charge_point.cpp

```diff
@@ -425,8 +425,43 @@ void ChargePoint::init_websocket() {
 
     this->websocket = std::make_unique<Websocket>(connection_options, this->pki_handler, this->logging);
     this->websocket->register_connected_callback([this](const int security_profile) {
+        
+        //handle offline threshold 
+        // Get the current time point using steady_clock
+        std::chrono::steady_clock::duration offline_duration = std::chrono::abs(std::chrono::steady_clock::now() - time_disconnected);
+
+        if (offline_duration >= offline_threshold_value) {
+            is_offline_threshold_expired = true;
+        }
+
         this->message_queue->resume();
         this->websocket_connection_status = WebsocketConnectionStatusEnum::Connected;
+
+        if (this->registration_status == RegistrationStatusEnum::Accepted) {
+
+            // B04.FR.01
+            // If offline period exceeds offline threshold then send the status notification for all connectors
+            if (is_offline_threshold_expired) {
```

Comment:
What about?
```suggestion
            auto offline_duration = std::chrono::steady_clock::now() - time_disconnected;

            if (offline_duration >= offline_threshold_value) {
```

Then you don't need `is_offline_threshold_expired` anymore and you could also use the threshold value directly without storing anything?